Environment.py

```diff
@@ -362,23 +411,22 @@ def step(self, action: Action) -> tuple[State, float, bool, Info]:
             assert not self.is_done, "Environment is finished."
 
             # Convert requests to epoch instance indices.
-            req2idx = {r: i for i, r in enumerate(self.ep_inst["request_idx"])}
+            req2idx = {r: i for i, r in enumerate(self.ep_inst.request_idx)}
             idx_sol = [[req2idx[req] for req in route] for route in action]
 
             # Check that all must-dispatch requests are dispatched.
-            must = np.flatnonzero(self.ep_inst["must_dispatch"])
+            must = np.flatnonzero(self.ep_inst.must_dispatch)
             dispatched = {req for route in idx_sol for req in route}
 
             msg = "Not all must-dispatch requests are dispatched."
             assert set(must).issubset(dispatched), msg
 
             # Check that the (static) solution is feasible.
-            cost = validate_static_solution(
-                self.ep_inst, idx_sol, allow_skipped_customers=True
-            )
+            cost = validate_static_solution(self.ep_inst, idx_sol)
+
         except AssertionError as error:
             self.is_done = True
-            return ({}, float("inf"), self.is_done, {"error": str(error)})
+            raise RuntimeError(f"Invalid action. Error: {error}.")
```

Comment:
We can be more explicit by raising from the exception here:
```suggestion
            raise RuntimeError(f"Invalid action.") from error
```
This prints the full traceback of `error` as well as the actual `RuntimeError` if the runtime error is not handled. That's much more helpful if this ever needs to be debugged.